app/getPairNumberFromColor.js

```diff
@@ -0,0 +1,43 @@
+const FetchMajorMinorColors = require("./FetchMajorMinorColors");
+
+class ProcessPairNumberFromColor {
+  constructor() {
+    this.colorList = new FetchMajorMinorColors();
+    this.MajorColorNames = this.colorList.getMajorColorList();
+    this.MinorColorNames = this.colorList.getMinorColorList();
+    this.minorColorIndex = -1;
+    this.majorColorIndex = -1;
+  }
+
+  updateMajorColorIndex(pair) {
+    for (let i = 0; i < this.MajorColorNames.length; i++) {
+      if (this.MajorColorNames[i] == pair.majorColor) {
+        this.majorColorIndex = i;
+        break;
+      }
+    }
+  }
+
+  updateMinorColorIndex(pair) {
```

Comment:
Notice repeated patterns of iteration and comparison. Move this functionality into a function and call it in both places, to remove the duplication.